Microsoft.Toolkit.Uwp/Extensions/DispatcherQueueExtensions.cs

```diff
@@ -0,0 +1,251 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Windows.System;
+
+#nullable enable
+
+namespace Microsoft.Toolkit.Uwp.Extensions
+{
+    /// <summary>
+    /// Helpers for executing code in a <see cref="DispatcherQueue"/>.
+    /// </summary>
+    public static class DispatcherQueueExtensions
+    {
+        /// <summary>
+        /// Invokes a given function on the target <see cref="DispatcherQueue"/> and returns a
+        /// <see cref="Task"/> that completes when the invocation of the function is completed.
+        /// </summary>
+        /// <param name="dispatcher">The target <see cref="DispatcherQueue"/> to invoke the code on.</param>
+        /// <param name="function">The <see cref="Action"/> to invoke.</param>
+        /// <param name="priority">The priority level for the function to invoke.</param>
+        /// <returns>A <see cref="Task"/> that completes when the invocation of <paramref name="function"/> is over.</returns>
+        /// <remarks>If the current thread has access to <paramref name="dispatcher"/>, <paramref name="function"/> will be invoked directly.</remarks>
+        public static Task EnqueueAsync(this DispatcherQueue dispatcher, Action function, DispatcherQueuePriority priority = DispatcherQueuePriority.Normal)
+        {
+            // Run the function directly when we have thread access.
+            // Also reuse Task.CompletedTask in case of success,
+            // to skip an unnecessary heap allocation for every invocation.
+            if (dispatcher.HasThreadAccess)
```

Comment:
HasThreadAccess was added only on 18362, so, unfortunately, we still need to check for its presence:
https://docs.microsoft.com/en-us/uwp/api/windows.system.dispatcherqueue.hasthreadaccess?view=winrt-19041#Windows_System_DispatcherQueue_HasThreadAccess
If this code is called on our min (17763), this code will throw.
What you could do, that is better than my old solution, is to cache the ```ApiInformation.IsMethodPresent``` since it will not change during the execution of the app. But you still need to check before you call it.

Reply:
Ooooh, my bad, I got confused by looking at the `HasThreadAccess` for the `CoreDispatcher` class 😅

Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3498/commits/25ed2d773aa77128a1991b38b2fcae180462aee8 with the value being cached as well in a statically initialized field, good idea! 🚀